### DISCUSSION TITLE
How can I contribute to this Project?

### DISCUSSION THREAD

**Discussion by NiyonAlex** [LMMS/lmms#7634] [General]
Who can provide me permission to contribute to this project?

**Comment by khoidauminh** [comment]
You're free to contribute to the project without permission!
The https://github.com/LMMS/lmms/wiki contains guides on how to start

**Comment by NiyonAlex** [comment]
When I push my branch, this window alerted.
![image](https://github.com/user-attachments/assets/38ddce65-508b-4ac6-9a7a-df8005aa979d)

How can I solve this?

**Reply by regulus79** [reply]
You need to push to your own fork of LMMS on github, and then make a pull request to merge it into the official repo.

**Comment by NiyonAlex** [comment]
I mean I want to contribute my skills to your official repo not my fork repo.
Do you get it?

**Reply by regulus79** [reply]
Yes, I understand. You want to contribute your code to the official LMMS repository so that all users can benefit.

Very few people actually have write access to the official LMMS repo. Instead, the way everyone (including me) contributes to the project is by first coding the feature/bugifx locally, then pushing to their own fork, and then opening a pull request on the LMMS repo. The developers will look at the pull request, review your code, maybe request changes, and then once they think it looks good, they will merge your code into the official repo.

**Comment by NiyonAlex** [comment]
Hi, I made a pull request to your repos.
I just fix 3 minor typos in this contribution. 
https://github.com/LMMS/lmms/pull/7635
I will contribute more.
